app/components/markdown/latex_code_block/latex_code_block.js

```diff
@@ -0,0 +1,204 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import {Keyboard, View, Text, StyleSheet, Platform} from 'react-native';
+import MathView from 'react-native-math-view';
+
+import {goToScreen} from '@actions/navigation';
+import FormattedText from '@components/formatted_text';
+import TouchableWithFeedback from '@components/touchable_with_feedback';
+import {splitLatexCodeInLines} from '@utils/latex';
+import {getDisplayNameForLanguage} from '@utils/markdown';
+import {preventDoubleTap} from '@utils/tap';
+import {changeOpacity, makeStyleSheetFromTheme} from '@utils/theme';
+
+import MarkdownCodeBlock from '../markdown_code_block/markdown_code_block';
+
+const MAX_LINES = 2;
+
+export default class LatexCodeBlock extends MarkdownCodeBlock {
```

Comment:
I had a look at changing the files to typescript but it seemed a lot more complex than the webapp because I had no real example to go by. But if I have some time left in the following week I will have a look at that.

In the mean time I'll add the `propTypes`

Reply:
Works for me. It's been a while since I've worked on the mobile app, and I know most of the TypeScript migration effort has been on the web-side, so I'm not surprised that it's trickier to do here.